modules/dnn/src/model.cpp

```diff
@@ -637,6 +637,335 @@ void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
         CV_Error(Error::StsNotImplemented, "Unknown output layer type: \"" + lastLayer->type + "\"");
 }
 
+class YOLODetectionModel_Impl : public DetectionModel_Impl
+{
+    // Define the enum for YOLO versions.
+    // NOTE: yolox here assumes that archor boxes
+    // decoding is already in the onnx model
+    enum class YoloVersion {
+        YOLOVOC = 0,
+        YOLOv3  = 3,
+        YOLOv4  = 4,
+        YOLOv5  = 5,
+        YOLOv6  = 6,
+        YOLOv7  = 7,
+        YOLOv8  = 8,
+        YOLOx   = 9,
+    };
+
+    ImagePaddingMode paddingMode;
+    YoloVersion yoloVersion;
+    float padValue;
+    float Height, Width;
```

Comment:
```suggestion
    Size frameSize;
```

Reply:
when I use Size instead of floats model  predictions start drifting for some unknown reason

Reply:
Probably because of integer division